File: frontend/src/sqhemas/dto/staff.ts
import { StatusCar } from "../enums/enums"
import { UserData } from "./user"

export interface CallsStatusDTO{
    id: number
    description: string
}

export interface CallsDTO{
    id: number
    time_call: Date
    type: string
    status_rel: CallsStatusDTO
}

export interface CallsDispatcher extends CallsDTO{
    dispatcher_id: number
}

export interface CallsMedStuffDTO extends CallsDTO{
    brigade_id: number
}

export interface CarDTO{
    id: number
    status: StatusCar
    location: string
}


export interface CompositionBrigadeDTO{
    staff_rel: UserData[]
}


export interface BrigadesDTO{
    id: number
    composition_rel: CompositionBrigadeDTO
    car_rel: CarDTO
}

File: db/sqhemas/dto/staff.py
from datetime import datetime
from pydantic import BaseModel

from db.sqhemas.dto.user import UserData
from db.sqhemas.enums.status_car import StatusCar


class CallsStatusDTO(BaseModel):
    id: int
    description: str


class CallsDTO(BaseModel):
    id: int
    time_call: datetime
    type: str
    status_rel: CallsStatusDTO


class CallsDispatcher(CallsDTO):
    dispatcher_id: int


class CallsMedStuffDTO(CallsDTO):
    brigade_id: int


class CarDTO(BaseModel):
    id: int
    status: StatusCar
    location: str


class CompositionBrigadeDTO(BaseModel):
    driver_rel: UserData
    paramedic_rel: UserData
    nurse_rel: UserData
    orderly_rel: UserData


class BrigadesDTO(BaseModel):
    id: int
    composition_rel: CompositionBrigadeDTO
    car_rel: CarDTO
